docusaurus/docs/dev-docs/plugins/users-permissions.md

```diff
@@ -922,6 +922,41 @@ The use of `ngrok` is not needed.
 Your configuration is done.
 Launch the backend and the [react login example app](https://github.com/strapi/strapi-examples/tree/master/examples/login-react), go to `http://localhost:3000` and try to connect to the provider your configured.
 
+##### Creating a custom provider
+
+<!-- TODO: add path of file to create 👇 -->
+You can also create your own custom provider, creating a … file and using the following code example adjusted to your needs:
+
+```js title="/path/to/the/file.js"
+module.exports = {
+  register({ strapi }) {
+    strapi
+      .plugin("users-permissions")
+      .service("providers-registry")
+      .add("example-provider-name", {
+        icon: "",
+        enabled: true,
+        grantConfig: {
+          key: "",
+          secret: "",
+          callback: `${strapi.config.server.url}/auth/test-provider/callback`,
+          scope: ["email"],
+          authorize_url: "https://awesome.com/authorize",
+          access_url: "https://awesome.com/token",
+          oauth: 2,
+        },
+        async authCallback({ accessToken, providers, purest }) {
```

Comment:
I don't know if we mention purest somewhere else but woudl make sense to also link to it